homeassistant/components/telegram_bot/__init__.py

```diff
@@ -73,11 +206,96 @@ def async_setup_platform(p_type, p_config=None, discovery_info=None):
+            _LOGGER.debug('NEW telegram_message "%s": %s', msgtype, kwargs)
+
+            if msgtype == SERVICE_SEND_MESSAGE:
+                yield from notify_service.async_send_message(**kwargs)
+            elif msgtype == SERVICE_SEND_PHOTO:
+                yield from notify_service.async_send_file(True, **kwargs)
+            elif msgtype == SERVICE_SEND_DOCUMENT:
+                yield from notify_service.async_send_file(False, **kwargs)
+            elif msgtype == SERVICE_SEND_LOCATION:
+                yield from notify_service.async_send_location(**kwargs)
+            elif msgtype == SERVICE_ANSWER_CALLBACK_QUERY:
+                yield from notify_service.async_answer_callback_query(**kwargs)
+            else:
+                yield from notify_service.async_edit_message(msgtype, **kwargs)
+
+        # Register notification services
+        hass.services.async_register(
+            DOMAIN, SERVICE_SEND_MESSAGE, async_send_telegram_message,
+            descriptions.get(SERVICE_SEND_MESSAGE),
+            schema=SERVICE_SCHEMA_SEND_MESSAGE)
+
+        hass.services.async_register(
+            DOMAIN, SERVICE_SEND_PHOTO, async_send_telegram_message,
+            descriptions.get(SERVICE_SEND_PHOTO),
+            schema=SERVICE_SCHEMA_SEND_FILE)
+        hass.services.async_register(
+            DOMAIN, SERVICE_SEND_DOCUMENT, async_send_telegram_message,
+            descriptions.get(SERVICE_SEND_DOCUMENT),
+            schema=SERVICE_SCHEMA_SEND_FILE)
+        hass.services.async_register(
+            DOMAIN, SERVICE_SEND_LOCATION, async_send_telegram_message,
+            descriptions.get(SERVICE_SEND_LOCATION),
+            schema=SERVICE_SCHEMA_SEND_LOCATION)
+
+        hass.services.async_register(
+            DOMAIN, SERVICE_EDIT_MESSAGE, async_send_telegram_message,
+            descriptions.get(SERVICE_EDIT_MESSAGE),
+            schema=SERVICE_SCHEMA_EDIT_MESSAGE)
+        hass.services.async_register(
+            DOMAIN, SERVICE_EDIT_CAPTION, async_send_telegram_message,
+            descriptions.get(SERVICE_EDIT_CAPTION),
+            schema=SERVICE_SCHEMA_EDIT_CAPTION)
+        hass.services.async_register(
+            DOMAIN, SERVICE_EDIT_REPLYMARKUP, async_send_telegram_message,
+            descriptions.get(SERVICE_EDIT_REPLYMARKUP),
+            schema=SERVICE_SCHEMA_EDIT_REPLYMARKUP)
+
+        hass.services.async_register(
+            DOMAIN, SERVICE_ANSWER_CALLBACK_QUERY, async_send_telegram_message,
+            descriptions.get(SERVICE_ANSWER_CALLBACK_QUERY),
+            schema=SERVICE_SCHEMA_ANSWER_CALLBACK_QUERY)
+
+        return True
+
     setup_tasks = [async_setup_platform(p_type, p_config) for p_type, p_config
                    in config_per_platform(config, DOMAIN)]
 
     if setup_tasks:
-        yield from asyncio.wait(setup_tasks, loop=hass.loop)
+        # Only one platform (webhooks or polling) is allowed
+        yield from asyncio.wait(setup_tasks[:1], loop=hass.loop)
```

Comment:
If only 1 is allowed, we should only setup 1. So line 293 should be updated.

However, if we only allow 1, we should not have a `PLATFORM_SCHEMA` to begin with but instead have a `CONFIG_SCHEMA`.